packages/sources/s3reader/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  TestAdapter,
+  setEnvVariables,
+} from '@chainlink/external-adapter-framework/util/testing-utils'
+import { GetObjectCommand } from '@aws-sdk/client-s3'
+
+jest.mock('@aws-sdk/client-s3', () => {
+  const originalModule = jest.requireActual('@aws-sdk/client-s3')
+  return {
+    ...originalModule,
+    S3Client: jest.fn().mockImplementation(() => {
+      return {
+        send: jest.fn().mockImplementation(async (command: GetObjectCommand) => {
+          if (command.input.Bucket != 's3_bucket') {
+            throw new Error('Error: The specified bucket does not exist.')
+          }
+          if (command.input.Key != 'correct/path/file.csv') {
+            throw new Error('Error: The specified key does not exist.')
+          }
+          return {
+            Body: {
+              transformToString: () =>
+                'Date,01/01/01,\nName,Value,Other\nRowOne,123,Other1\nRowTwo,456,Other2\nRowThree,789,Other3',
```

Comment:
I think in the past I have worked with csvs, which are formatted like:

'ColHeader1,ColHeader2,ColHeader3\nRow1Value1,Row1Value2,Row1Value3\nRow2Value1,Row2Value2,Row2Value3...'

Can we support and test for this?

Reply:
I see that is the format actually. The first row of the csv threw me off!

Reply:
great!